code/modules/events/ailments.dm

```diff
@@ -0,0 +1,19 @@
+/datum/event/ailments/start()
+	var/list/candidates = list()
+	for(var/mob/living/carbon/human/H in GLOB.living_mob_list_)
+		if(H.client && !length(H.stasis_sources))
+			candidates += H
+	var/create_ailments = min(length(candidates), rand(1,3))
+	while(length(candidates) && create_ailments)
+		var/mob/living/carbon/human/H = pick_n_take(candidates)
+		var/list/organs = (H.organs|H.internal_organs)
+		while(length(organs))
+			var/obj/item/organ/O = pick_n_take(organs)
```

Comment:
It's a lot faster to shuffle this once and then loop over it instead of pick_n_taking repeatedly.